src/Adapter/CartRule/CommandHandler/AddCartRuleHandler.php

```diff
@@ -142,9 +138,9 @@ private function fillCartRuleActionsFromCommandData(CartRule $cartRule, AddCartR
             null;
 
         // Legacy reduction_tax property is true when it's tax included, false when tax excluded.
-        $cartRule->reduction_tax = null !== $amountDiscount ? !$amountDiscount->isTaxExcluded() : null;
+        $cartRule->reduction_tax = null !== $amountDiscount ? $amountDiscount->isTaxIncluded() : null;
 
-        $cartRule->reduction_percent = null !== $percentageDiscount ? $percentageDiscount->getPercentage() : null;
+        $cartRule->reduction_percent = null !== $percentageDiscount ? (string) $percentageDiscount->getPercentage() : null;
```

Comment:
```suggestion
        $cartRule->reduction_percent = null !== $percentageDiscount ? (string) (float) $percentageDiscount->getPercentage() : null;
```
This is supposed to be a float no?